feat: improve deployment pattern (#344)
open-feature/open-feature-operator: docs/getting_started.md

        targeting: {}
```

### Reference the deployed FeatureFlagConfiguration within FlagSourceConfiguration.

> The `FlagSourceConfiguration` defined below can be used to assign our `FeatureFlagConfiguration`, as well as any other configuration settings, to the injected sidecars. In this example we are also setting the port exposed by our injected container.

Reviewer: ```suggestion
The `FlagSourceConfiguration` defined below can be used to assign the `FeatureFlagConfiguration`, as well as any other configuration settings, to the injected sidecars. In this example, the port exposed by the injected container is also set.
```

```yaml
apiVersion: core.openfeature.dev/v1alpha2
kind: FlagSourceConfiguration
metadata:
  name: flagsourceconfiguration-sample
spec:
  syncProviders:
  - source: featureflagconfiguration-sample
    provider: kubernetes
  port: 8080
```

### Reference the deployed FlagSourceConfiguration within a Deployment spec annotation.

> In this example, a `Deployment` containing a `busybox-curl` container is created. In the example below, the `metadata.annotations` object contains the required annotations for the operator to correctly configure and inject the `flagd` sidecar into each deployed `Pod`. The documentation for these annotations can be found [here](./annotations.md).

Reviewer: ```suggestion
In this example, a `Deployment` containing a `busybox-curl` container is created. In the configuration below, the `metadata.annotations` object contains the required annotations for the operator to correctly configure and inject the `flagd` sidecar into each deployed `Pod`. The documentation for these annotations can be found [here](./annotations.md).
```
